src/containers/pages/MDAPoint/ClientListView/index.tsx

```diff
@@ -0,0 +1,159 @@
+  files: File[] | null;
+  serviceClass: typeof OpenSRPService;
+}
+/** default props for ClientListView component */
+export const defaultClientListViewProps: ClientListViewProps = {
+  fetchFilesActionCreator: fetchFiles,
+  files: null,
+  serviceClass: OpenSRPService,
+};
+/**
+ * Builds list view table data
+ * @param {File[] } rowData file data coming from opensrp/history endpoint
+ */
+export const buildListViewData: (rowData: File[]) => ReactNode[][] | undefined = rowData => {
+  return rowData.map((row: File, key: number) => {
+    const { url, fileName } = row;
+    return [
+      <p key={key}>
+        {fileName} &nbsp;
+        {/* tslint:disable-next-line jsx-no-lambda */}
+        <a href="#" onClick={() => handleDownload(url, fileName)}>
+          (Downloads)
+        </a>
+      </p>,
+      row.providerID,
+      row.uploadDate,
+    ];
+  });
+};
+
+export const ClientListView = (props: ClientListViewProps & RouteComponentProps) => {
+  React.useEffect(() => {
+    if (!(props.files && props.files.length)) {
+      /**
+       * Fetch files incase the files are not available e.g when page is refreshed
+       */
+      loadFiles().catch(err => displayError(err));
+      // fetchFilesActionCreator(uploadedStudentsLists);
+    }
+    /**
+     * We do not need to re-run since this effect doesn't depend on any values from api yet
+     */
+  }, []);
+  /** Overide renderRows to render html inside td */
+  let listViewProps;
+  if (props.files && props.files.length) {
+    listViewProps = {
+      data: buildListViewData(props.files),
+      headerItems: ['File Name', 'Owner', 'Upload Date'],
+      tableClass: 'table table-bordered',
+    };
+  }
+  /** Load Modal once we hit this route */
+  if (props.location.pathname === '/clients/students/upload') {
+    return <ClientUpload />;
+  }
+  /** props to pass to the headerBreadCrumb */
+  const breadcrumbProps: BreadCrumbProps = {
+    currentPage: {
+      label: ENABLE_MDA_POINT ? STUDENTS_TITLE : CLIENTS_TITLE,
+      url: STUDENTS_LIST_URL,
+    },
+    pages: [],
+  };
+  const homePage = {
+    label: `${HOME}`,
+    url: `${HOME_URL}`,
+  };
+  breadcrumbProps.pages = [homePage];
+
+  // props for the link displayed as button: used to add new practitioner
+  const csvUploadButtonProps = {
+    text: ADD_NEW_CSV,
+    to: UPLOAD_STUDENT_CSV_URL,
+  };
+  return (
+    <div>
+      <Helmet>
+        <title>{STUDENTS_TITLE}</title>
+      </Helmet>
+      <HeaderBreadcrumb {...breadcrumbProps} />
+      <Row id="header-row">
+        <Col className="xs">
+          <h3 className="mb-3 mt-5 page-title">
+            {ENABLE_MDA_POINT ? UPLOADED_STUDENT_LISTS : UPLOADED_CLIENT_LISTS}
+          </h3>
+        </Col>
+        <Col className="xs">
+          <LinkAsButton {...csvUploadButtonProps} />
+        </Col>
+      </Row>
+      <Row id="table-row">
+        <Col>
+          {listViewProps && props.files && props.files.length ? (
```

Comment:
I do not think this is a good check for whether loading is happening.  For one thing, in the event the API does not return any files you will show a loading indicator for no reason.  I think you should do this when you are calling `loadFiles` and in accordance with the ideas here https://github.com/onaio/reveal-frontend/issues/806

Reply:
https://github.com/onaio/reveal-frontend/issues/890